### ISSUE TITLE
Sync Batch Normalization?

### ISSUE THREAD

**Issue by weiminggao** [bytedance/byteps#284]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by ymjiang** [comment]
Sorry, it is not supported yet. We may look into this in the future.

**Comment by bobzhuyb** [comment]
@weiminggao I believe that you can use push_pull to implement global syncBN if you don't mind writing a little code.

**Comment by weiminggao** [comment]
Can you describe more detail? How to use push_pull to implement global syncBN?
I use the tf.layers.BatchNormalzation() api.

**Comment by weiminggao** [comment]
I find syncBN need to exchange the all moving_mean and moving_var , and then compute the real moving_mean and moving_var. 
![image](https://user-images.githubusercontent.com/42035017/89282795-a3c45d80-d67e-11ea-8be2-4f6171850968.png)

**Comment by bobzhuyb** [comment]
In distributed training, \sigma can be implemented by Horovod's all-reduce or BytePS's push_pull  (these two are equivalent in synchronized data parallel)
